sanic/testing.py

```diff
@@ -203,7 +207,7 @@ def __init__(
 
         self.app = app
 
-        dispatch = SanicASGIDispatch(app=app, client=(ASGI_HOST, PORT))
+        dispatch = SanicASGIDispatch(app=app, client=(ASGI_HOST, PORT or 0))
```

Comment:
Why 0 ?

Reply:
Typing says it needs to be int, so `None` causes a failure. Either the type hint should be changed to `Optional[int]` or 0 be used instead but I did not look at ASGI code to see how this is actually handled. In any case, 0 is the standard way to request for a free random port.

Reply:
Actually all of Sanic should probably be changed to accept 0 port. Currently entering zero gives port 8000. But that is a potentially breaking change, and for another PR even if implemented.